### PR TITLE
Extract queries into separate package and add new ones 

### PR DESCRIPTION
**Pull request by vitsalis**
In this PR we extract the querying functionality of the RPC Client into a separate package called `query`. The ultimate goal is to be able to facilitate querying Babylon without using the `ChainClient` instance, but instead a simple `grpc` client instance. This will be worked on in future PRs.

Further, this PR:
- Refactors all query endpoints to respond with QueryResponse types
- Adds a bunch of extra query endpoints needed by the Babylon API
- Removes `Must*` methods. This methods rely on a retrying mechanism which has no place in this client. The dependents can do retrying themselves if they want to. Further, this alleviates the need for providing the retry sleep times as arguments to the client, which might rarely get used (if someone does not use the `Must*` methods)

This PR will be followed with:
- The creation of an `0.6.0rc1` tag
- The modification of the vigilante to depend on this new version and the update of the vigilante
- Updating the babylon API to depend on the new changes

### CODE REVIEW

**Comment by SebastianElvis** [comment]
That's a great effort! I have an idea on further removing the need of `GetDefaultQueryContext`. If in the future we found a way to replace `ChainClient` with something else in `NewQueryClient` then all query functions will not need a `ChainClient`. I detailed the approach below and maybe you could have a look.

The `query.GetQueryContext` is defined as follows

```go
// GetQueryContext returns a context that includes the height and uses the timeout from the config
func (q *Query) GetQueryContext() (context.Context, context.CancelFunc) {
	timeout, _ := time.ParseDuration(q.Client.Config.Timeout) // Timeout is validated in the config so no error check
	ctx, cancel := context.WithTimeout(context.Background(), timeout)
	strHeight := strconv.Itoa(int(q.Options.Height))
	ctx = metadata.AppendToOutgoingContext(ctx, grpctypes.GRPCBlockHeightHeader, strHeight)
	return ctx, cancel
}
```

where `q.Options` is `query.DefaultOptions()`, which is defined as follows

```go
func DefaultOptions() *QueryOptions {
	return &QueryOptions{
		Pagination: &query.PageRequest{
			Key:        []byte(""),
			Offset:     0,
			Limit:      1000,
			CountTotal: true,
		},
		Height: 0,
	}
}
```

So we can get the default context below, which is independent from `ChainClient`:

```go
func GetDefaultContext(cfg *BabylonConfig) (context.Context, context.CancelFunc) {
	queryOptions := &query.QueryOptions{
		Pagination: &query.PageRequest{
			Key:        []byte(""),
			Offset:     0,
			Limit:      1000,
			CountTotal: true,
		},
		Height: 0,
	}

	timeout, _ := time.ParseDuration(cfg.Timeout) // Timeout is validated in the config so no error check
	ctx, cancel := context.WithTimeout(context.Background(), timeout)
	strHeight := strconv.Itoa(int(queryOptions.Height))
	ctx = metadata.AppendToOutgoingContext(ctx, grpctypes.GRPCBlockHeightHeader, strHeight)
	return ctx, cancel
}
```

What do you think?

**Review by SebastianElvis — APPROVED** [review]
All good from my side! 💯 Left a comment on further cleaning the query functions, maybe we could consider it.

**Comment by vitsalis** [comment]
Great suggestion! Although, the default query function that you propose still needs access to the Babylon configuration for `cfg.Timeout` and therefore still depends on `client` as it maintains the `cfg` attribute. I suggest we leave this for a future refactoring.

File: client/client.go [hunk 1]
```diff
@@ -13,10 +15,6 @@ var _ BabylonClient = &Client{}
 type Client struct {
 	*lensclient.ChainClient
 	cfg *config.BabylonConfig
-
-	// retry attributes
-	retrySleepTime    time.Duration
-	maxRetrySleepTime time.Duration
 }
 
 func New(cfg *config.BabylonConfig, retrySleepTime, maxRetrySleepTime time.Duration) (*Client, error) {
```
Comment (gitferry) [inline client/client.go:20]:
```suggestion
func New(cfg *config.BabylonConfig) (*Client, error) {
```

**Review by gitferry — APPROVED** [review]
Great refactoring. One bit is that do we want to run `make mock-gen` to generate mocks here?

**Pull request merged**